.github/workflows/sonar-analysis-on-merge.yml

```diff
@@ -0,0 +1,39 @@
+name: Sonar Analysis on Merge
```

Comment:
What's the reason for having this in both the `checks.yml` and `sonar-analysis-on-merge.yml` workflows?

Reply:
Commented out on the checks.yml. Initially tested with checks.yml, but keeping it in separate file will make it more clear.

Reply:
Removed the checks.yml changes from the commit and only using sonar-analysis-on-merge.yml